test/system/home_test.rb

```diff
@@ -14,32 +14,101 @@ class HomeTest < ApplicationSystemTestCase
   end
 
-  test 'GET / without discord_account' do
+  test 'verify message presence of avatar registration' do
     visit_with_auth '/', 'hajime'
-    within('.card-list__item-link.is-discord_account') do
-      assert_text 'Discordアカウントを登録してください。'
-    end
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'ユーザーアイコンを登録してください。'
+
+    visit '/current_user/edit'
+    file_path = Rails.root.join('test/fixtures/files/users/avatars/default.jpg')
+    attach_file 'user[avatar]', file_path, make_visible: true
+    click_button '更新する'
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'ユーザーアイコンを登録してください。'
   end
 
-  test 'GET / with discord_account' do
-    user = users(:hajime)
-    user.update!(discord_account: 'hajime#1111')
-    visit_with_auth '/', 'hajime'
-    assert_no_selector '.card-list__item-link.is-discord_account'
+  test 'verify message presence of tags registration' do
+    visit_with_auth '/', 'hatsuno'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'タグを登録してください。'
+
+    visit '/current_user/edit'
+    tag_input = find('.ti-new-tag-input')
+    tag_input.set '猫'
+    tag_input.native.send_keys :return
+    click_button '更新する'
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'タグを登録してください。'
+  end
+
+  test 'verify message presence of after_graduation_hope registration' do
+    visit_with_auth '/', 'hatsuno'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'フィヨルドブートキャンプを卒業した自分はどうなっていたいかを登録してください。'
+
+    visit '/current_user/edit'
+    fill_in 'user[after_graduation_hope]', with: 'IT ジェンダーギャップ問題を解決するアプリケーションを作る事業に、エンジニアとして携わる。'
+    click_button '更新する'
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'フィヨルドブートキャンプを卒業した自分はどうなっていたいかを登録してください。'
+  end
+
+  test 'verify message presence of blog_url registration' do
+    users(:hatsuno).update!(blog_url: '') # 確認のために削除
+    visit_with_auth '/', 'hatsuno'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'ブログURLを登録してください。'
+
+    visit '/current_user/edit'
+    fill_in 'user[blog_url]', with: 'http://hatsuno.org'
+    click_button '更新する'
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'ブログURLを登録してください。'
+  end
+
+  test 'not show messages of required field' do
+    user = users(:hatsuno)
+    # hatsuno の未入力項目を登録
+    user.update!(
+      tag_list: ['猫'],
+      after_graduation_hope: 'IT ジェンダーギャップ問題を解決するアプリケーションを作る事業に、エンジニアとして携わる。',
+      discord_account: 'hatsuno#1234'
+    )
+    path = Rails.root.join('test/fixtures/files/users/avatars/hatsuno.jpg')
+    user.avatar.attach(io: File.open(path), filename: 'hatsuno.jpg')
+
+    visit_with_auth '/', 'hatsuno'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text '未入力の項目'
```

Comment:
テスト意図としては以下のように読み取りましたが、いかがでしょうか〜
```suggestion
    visit_with_auth '/', 'hatsuno'
    # 未入力項目の登録前
    assert_selector 'h2.card-header__title', text: '未入力の項目'
    assert_no_selector 'h2.card-header__title', text: '卒業後のゴール'
    # ：
    # ：（中略：登録処理）
    # ：
    # 未入力項目の登録後
    assert_no_selector 'h2.card-header__title', text: '未入力の項目'
    assert_selector 'h2.card-header__title', text: '卒業後のゴール'
```

Reply:
現在のテストデータだと提案していただいたテストも通るのですが、「未入力の項目」があっても 「卒業後のゴール」は表示されるので、このテストは不適切なものになると思います。（`hatsuno` の `after_graduation_hope` を削除してからテストすると落ちます）

テストの意図は、対象のプロフィール項目が全て登録されているユーザーでログインした際に「未入力の項目」が表示されないことだったので、「未入力の項目」の表示の変化は対象としていませんでした。
「未入力の項目」が表示されていることは（暗黙的にはなってしまうのですが）他のテストで示唆されており、特別テストする必要はないと考えています。

Reply:
たしかに、下記のとおりでした〜🙏　切り替わる仕様と思っていたら別物だったのですね。このままで大丈夫です👍

> 「未入力の項目」があっても 「卒業後のゴール」は表示される